src/utils.rs

```diff
@@ -69,15 +69,16 @@ pub async fn send_newest_updates(
     context: &Context,
     msg_id: MsgId,
     db: &mut SqliteConnection,
-    serial: i32,
+    serial: u32,
 ) -> anyhow::Result<()> {
-    let app_infos: Vec<_> = db::get_active_app_infos_since(db, serial)
-        .await?
-        .into_iter()
-        .collect();
-
+    let app_infos: Vec<_> = db::get_active_app_infos_since(db, serial).await?;
+    let removed = db::get_inactive_app_infos_since(db, serial).await?;
     let serial = db::get_last_serial(db).await?;
-    let resp = ShopResponse::Update { app_infos, serial };
+    let resp = ShopResponse::Update {
+        app_infos,
+        removed: removed.into_iter().map(|app_info| app_info.id).collect(),
```

Comment:
what is this "removed" list of app_id's meant to signal to the store.xdc? 
i first thought that "removed" means that the whole app (with all versions) got removed but `get_inactive_app_infos_since` provides all old versions.  isn't "removed"  non-empty for a simply upgraded app (i.e. there was version 1, now there is version 2)?

Reply:
Yes it not empty as it removes the old appinfo and adds a new one (that for version 2)